syncer/streamer_controller.go

```diff
@@ -367,15 +366,15 @@ func (c *StreamerController) updateServerID(tctx *tcontext.Context) error {
 		// should never happened unless the master has too many slave
 		return terror.Annotate(err, "fail to get random server id for streamer controller")
 	}
+
+	c.RLock()
```

Comment:
Do we need `Lock` rather than `RLock` here?

Reply:
yes, it should use Lock, fix in bd851a0